components/merkledb/src/indexes/group.rs

```diff
@@ -106,11 +107,32 @@ where
     }
 }
 
+impl<T, K, I> Group<T, K, I>
+where
+    T: RawAccess,
+    K: BinaryKey + ?Sized,
+    I: FromAccess<T>,
+{
+    /// Returns an iterator over the keys from this group.
+    ///
+    /// **Note:** use this method carefully, because it obtains all available keys for this
+    /// group in `suffixes` method and stores them in `Vec`. This operation can consume a large
+    /// amount of RAM.
+    ///
+    /// **Note:** creating any new indexes while this iterator is in scope may lead to panic.
```

Comment:
Sorry for obtuse remarks: *This* implementation cannot panic in the described scenario since it does not borrow the metadata view (unlike the implementation in #1637, which I admittedly specified in a very vague manner).

Because this implementation still has memory overhead, I'd suggest to use the name `buffered_keys` (or `keys_buffered`) and return a `Vec<K::Owned>` directly.